subscription-service/src/main/kotlin/com/egm/stellio/subscription/config/WebClientConfig.kt

```diff
@@ -0,0 +1,53 @@
+package com.egm.stellio.subscription.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.PropertySource
+import org.springframework.security.oauth2.client.AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.InMemoryReactiveOAuth2AuthorizedClientService
+import org.springframework.security.oauth2.client.registration.ClientRegistration
+import org.springframework.security.oauth2.client.registration.InMemoryReactiveClientRegistrationRepository
+import org.springframework.security.oauth2.client.registration.ReactiveClientRegistrationRepository
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServerOAuth2AuthorizedClientExchangeFilterFunction
+import org.springframework.security.oauth2.core.AuthorizationGrantType
+import org.springframework.web.reactive.function.client.WebClient
+
+@PropertySource("classpath:application.properties")
+@Configuration
+class WebClientConfig {
```

Comment:
why don't you have a simple config like in https://www.baeldung.com/spring-webclient-oauth2#spring-security-5-support---the-client-credentials-flow?

Reply:
I don't use the function UnAuthenticatedServerOAuth2AuthorizedClientRepository() because it is deprecated
So I use other method. I see another one this morning. I'm going to test it

Reply:
They suggesting to use `AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager`